main.tf

```diff
@@ -127,7 +145,7 @@ resource "aws_db_instance" "rds_master" {
   username                   = "${var.rds_instance_root_user_name}"
   password                   = "${var.rds_instance_root_user_password}"
   db_subnet_group_name       = "${var.rds_master_id == "" ? aws_db_subnet_group.rds_private_subnet.name : ""}"
-  parameter_group_name       = "${aws_db_parameter_group.rds_params.name}"
+  parameter_group_name       = "${var.max_connections != ""  ? aws_db_parameter_group.rds_params_with_max_connections.name : aws_db_parameter_group.rds_params_without_max_connections.name }"
```

Comment:
Is it not possible to have just one parameter group here and set a default value for max_connections if not passed by the parent module ?
Which means its always `rds_params_with_max_connections` but value of max_connections has a default if not passed down